SYCL/AtomicRef/add.cpp

```diff
@@ -1,44 +1,154 @@
-// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t.out \
-// RUN: -Xsycl-target-backend=nvptx64-nvidia-cuda --cuda-gpu-arch=sm_60
+// Each combination of 64/32 bit atomic, relaxed/acquire/release/acq_rel
+// semantic order and sub_group/work_group/device/system scope is tested
+// separately. This is controlled by macros, defined by RUN commands. Defaults
+// (no macro for a group) are: 32 bit, relaxed and device.
+
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t.out -Xsycl-target-backend --cuda-gpu-arch=sm_60
+// RUN: %HOST_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
```

Comment:
Is the the compiled for a single target - NVIDIA GPU? If so, I suggest run on GPU device only.
I expect this test to be super-slow comparing to other tests in this suite, because we compile it many times. I would prefer to reduce # of compilations as much as possible.

700 of 757 test finish in less than 10 sec on AMD GPU:

[ 8.0s,10.0s) :: [*                                       ] :: [ 25/757]
[ 6.0s, 8.0s) :: [**********                              ] :: [198/757]
[ 4.0s, 6.0s) :: [**                                      ] :: [ 40/757]
[ 2.0s, 4.0s) :: [*                                       ] :: [ 22/757]
[ 0.0s, 2.0s) :: [*********************                   ] :: [415/757]

Reply:
Atomic tests used to have multiple files for each operation. With the changes in this PR I would have to add many more if I did not refactor them - so we have equivalent of what would be 24 files in one here, but it still requires separate RUN comands. See the conversation here: https://github.com/intel/llvm-test-suite/pull/534#discussion_r743537200